iceoryx_utils/include/iceoryx_utils/internal/relocatable_pointer/relative_ptr.hpp

```diff
@@ -253,34 +253,36 @@ class relative_ptr : public RelativePointer
         return *this;
     }
 
-    T& operator*()
+    template <typename SFINAE = T>
+    typename std::enable_if<!std::is_void<SFINAE>::value, SFINAE&>::type operator*()
     {
-        return *(static_cast<T*>(computeRawPtr()));
+        return *get();
     }
 
     T* operator->()
     {
-        return static_cast<T*>(computeRawPtr());
+        return get();
     }
 
-    const T& operator*() const
+    template <typename SFINAE = T>
+    typename std::enable_if<!std::is_void<SFINAE>::value, const SFINAE&>::type operator*() const
     {
-        return *(static_cast<T*>(computeRawPtr()));
+        return *get();
     }
 
     T* operator->() const
     {
-        return static_cast<T*>(computeRawPtr());
+        return get();
     }
 
     T* get() const
     {
-        return reinterpret_cast<T*>(RelativePointer::get());
+        return static_cast<T*>(computeRawPtr());
```

Comment:
I would use a reinterpret_cast here. Under certain conditions (involving multiple inheritance of the types and maybe others) a static_cast can actually change the address. reinterpret_cast will never do this.
I prefer reinterpret casts if we cast from void to some typed pointer.

Reply:
`computeRawPtr()` returns a `void*` already so performing a `static_cast` won't perform a change in address.